packages/noya-state/src/reducers/canvasReducer.ts

```diff
@@ -141,6 +147,65 @@ export function canvasReducer(
         draft.selectedObjects = [layer.do_objectID];
       });
     }
+    case 'addStopToGradient': {
+      const [, point] = action;
+      const pageIndex = getCurrentPageIndex(state);
+      const position = Selectors.getPercentageOfPointInGradient(state, point);
+
+      if (!state.selectedGradient) return state;
+
+      const { layerId, fillIndex, styleType } = state.selectedGradient;
+
+      const page = getCurrentPage(state);
+      const indexPath = Layers.findIndexPath(
+        page,
+        (layer) => layer.do_objectID === layerId,
+      );
+
+      if (!indexPath) return state;
+
+      return produce(state, (draft) => {
+        const layer = Layers.access(draft.sketch.pages[pageIndex], indexPath);
+
+        if (
+          (styleType === 'fill' &&
+            layer.style?.fills?.[fillIndex].fillType !==
+              Sketch.FillType.Gradient) ||
+          (styleType === 'border' &&
+            layer.style?.borders?.[fillIndex].fillType !==
+              Sketch.FillType.Gradient)
+        )
+          return state;
+
+        const gradientStops =
+          styleType === 'fill'
+            ? layer.style?.fills?.[fillIndex].gradient.stops
+            : layer.style?.borders?.[fillIndex].gradient.stops;
+
+        if (!gradientStops) return;
+
+        const gradient = gradientStops.map((g) => ({
+          color: sketchColorToRgba(g.color),
+          position: g.position,
+        }));
+
+        const color = rgbaToSketchColor(interpolateRgba(gradient, position));
+        gradientStops.push({
```

Comment:
Very minor: we can use `SketchModel.gradientStop({ color, position })` to create a gradient stop, which automatically adds the `_class` property